service-map/src/main/java/fi/mml/map/mapwindow/util/OskariLayerWorker.java

```diff
@@ -22,23 +14,26 @@
 import fi.nls.oskari.map.layer.OskariLayerServiceMybatisImpl;
 import fi.nls.oskari.map.layer.formatters.LayerJSONFormatter;
 import fi.nls.oskari.util.JSONHelper;
+import org.oskari.permissions.PermissionService;
+import org.oskari.permissions.PermissionServiceMybatisImpl;
+import org.oskari.permissions.model.PermissionSet;
+import org.oskari.permissions.model.PermissionType;
+import org.oskari.permissions.model.Resource;
+import org.oskari.permissions.model.ResourceType;
 
 /**
  * Worker class for rendering json objects from domain objects
  */
 public class OskariLayerWorker {
 
     public static final String KEY_LAYERS = "layers";
-
-    private static final String NO_PUBLICATION_PERMISSION = "no_publication_permission";
-    private static final String PUBLICATION_PERMISSION_OK = "publication_permission_ok";
-
-    private static final String DOWNLOAD_PERMISSION_OK = "download_permission_ok";
+    public static final String PUBLICATION_PERMISSION_OK = "publication_permission_ok";
+    public static final String DOWNLOAD_PERMISSION_OK = "download_permission_ok";
 
     private static Logger log = LogFactory.getLogger(OskariLayerWorker.class);
 
     private static OskariLayerService mapLayerService = new OskariLayerServiceMybatisImpl();
-    private static PermissionsService permissionsService = new PermissionsServiceIbatisImpl();
+    private static PermissionService permissionService = new PermissionServiceMybatisImpl();
```

Comment:
ComponentManager

Reply:
Using OskariComponentManager here breaks all the GetAppSetup*Tests. There's something funky going on with mocking stuff coming out of a ServiceLoader.